packages/block-library/src/button/edit.native.js

```diff
@@ -228,9 +240,34 @@ class ButtonEdit extends Component {
 
 	onLayout( { nativeEvent } ) {
 		const { width } = nativeEvent.layout;
-		const { marginRight } = styles.button;
-		const buttonSpacing = 2 * marginRight;
-		this.setState( { maxWidth: width - buttonSpacing } );
+		this.onSetMaxWidth( width );
+	}
+
+	onSetMaxWidth( width ) {
+		const { maxWidth } = this.state;
+		const { parentWidth } = this.props;
+		const { marginRight: spacing } = styles.button;
+
+		const isParentWidthChanged = maxWidth !== parentWidth;
+		const isWidthChanged = maxWidth !== width;
+
+		if ( parentWidth && ! width && isParentWidthChanged ) {
+			this.setState( {
+				maxWidth: parentWidth,
+			} );
+		} else if ( ! parentWidth && width && isWidthChanged ) {
+			this.setState( { maxWidth: width - spacing } );
+		}
+	}
+
+	onRemove() {
+		const { numOfButtons, onDelete, onReplace } = this.props;
+
+		if ( numOfButtons === 1 ) {
+			onDelete();
```

Comment:
Please see comments below

we can use `onDeleteBlock` after passing it in BlockEdit in `block.native.js`